app/src/main/java/com/sdp/swiftwallet/domain/repository/AuthenticatorFirebaseImpl.java

```diff
@@ -74,6 +82,29 @@ public Result signIn(String email, String password, Runnable success, Runnable f
         return Result.SUCCESS;
     }
 
+    @Override
+    public Result signUp(String username, String email, String password, Runnable success, Runnable failure) {
+        if (username == null || email == null || password == null) return Result.ERROR;
+
```

Comment:
I do not understand the difference between this Result.ERROR and the `Result.EMPTY_EMAIL` one, why wouldn't you throw the same kind of error that you throw in signIn ?

Reply:
It will be use later with the helper functions that check for input validity